### DISCUSSION TITLE
Ability to do multiple claims at once

### DISCUSSION THREAD

**Discussion by adam-635** [iotaledger/firefly#4345] [Features / Enhancements]
User should have an ability to claim multiple received transactions at once. They might be receiving those transactions programatically and have many of them. Asking them to manually claim each of them could make Firefly unusable.

I would actually argue that "claim all" has higher priority over "claim individual". Typically user will not care and just want to claim all of them. They might even prefer an option where it's done automatically in the background.

<img width="826" alt="Screen Shot 2022-08-25 at 8 29 25 PM" src="https://user-images.githubusercontent.com/130610/186811507-e3eb9b69-868e-44c8-8ab6-b6c74aa99478.png">

**Comment by adam-635** [comment]
This could be a process for many providers, see example of me claiming fees from just few transactions that have occurred during our public testing in past one day. The claiming runs for a long time so I think it's important part of the "claim all" user is presented with progress window.

https://user-images.githubusercontent.com/130610/186995274-93017fce-39e5-4637-a169-11094bf413b0.mov

There might be an argument to consolidate those fees and avoid this. This something we're trying to avoid as it has negative effect for two main reasons:
- we would have to start running our "own" custody wallet to consolidate. We're not in wallet business :)
- it could result in less transparent process on tangle

Let me know if you need more info.

**Comment by anton0k** [comment]
I was about to create the same feature request. Your proposal totally makes sense. I would also go for a claim all button but I am to any other good idea :-)

**Comment by begonaalvarezd** [comment]
As a trick, you can run an output consolidation which should claim all. In the wallet tab, top right corner of your balance 
- `...` menu 
- view balance breakdown
- minimize storage deposit
- consolidate

to my understanding, that can potentially claim all 🪄

**Reply by anton0k** [reply]
Thank you very much for the hint. For me that did not work. Did it for you @begonaalvarezd?

**Reply by adam-635** [reply]
@TangleAccountant can you try it if it works for your 1000s of unclaimed outputs?

**Reply by TangleAccountant** [reply]
This didn't help.

**Reply by begonaalvarezd** [reply]
this trick only works for expiration unlock condition, not time unlock condition 🙏🏼

**Reply by TangleAccountant** [reply]
I don't believe I'm having trouble with time unlock conditions. It's for SMR that needs to be claimed in order for the storage deposit to be returned to the sender. @adamunchained correct me if I'm wrong.
